### PR TITLE
ORV2-579 Backend Changes

### PR DESCRIPTION
**Pull request by praju-aot**
# Description

ORV2-579 - Incorporates the new roles defined in ORBC_VT_USER_AUTH_GROUP in the BE as well

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- npm run test
- npm run test:e2e

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

MOTI Dev DB needs to be reset prior to merging these changes

---

Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend - Vehicles](https://onroutebc-247-backend-vehicles.apps.silver.devops.gov.bc.ca/) available
[Frontend](https://onroutebc-247-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge-main.yml)

### CODE REVIEW

File: backend/vehicles/src/common/enum/user-auth-group.enum.ts
Comment (john-fletcher-aot) [inline backend/vehicles/src/common/enum/user-auth-group.enum.ts:1]: This is going to be a dynamic list of roles, I expect. Is there a way we can allow MOTI to update the table of auth groups in the future without needing to update the UserAuthGroup code as well?

Comment (praju-aot) [reply backend/vehicles/src/common/enum/user-auth-group.enum.ts:1]: In future, we can probably make these configurable and load to cache on startup.

Comment (john-fletcher-aot) [reply backend/vehicles/src/common/enum/user-auth-group.enum.ts:1]: We'll leave it as an enum for now and look into making it a dynamic list in a future merge

**Review by john-fletcher-aot — APPROVED** [review]
Keeping it as is for now, we'll look into options for making it dynamic (if it doesn't negatively impact other things) later on so the group list can be edited without code changes once in production.

**Pull request merged**